core/blockchain.go

```diff
@@ -511,6 +630,333 @@ const (
 	SideStatTy
 )
 
+// writeHeader writes a header into the local chain, given that its parent is
+// already known. If the total difficulty of the newly inserted header becomes
+// greater than the old known TD, the canonical chain is re-routed.
```

Comment:
"old known" -> "current" is more accurate, since it's not old yet, and may not be depending on logic in this function.